specs/simpleserialize.md

```diff
@@ -0,0 +1,302 @@
+
+This document specifies the general information for serializing and
+deserializing objects and data types.
+
+## ToC
+
+* [About](#about)
+* [Terminology](#terminology)
+* [Constants](#constants)
+* [Overview](#overview)
+   + [Serialize/Encode](#serializeencode)
+      - [uint: 8/16/24/32/64/256](#uint-816243264256)
+      - [Address](#address)
+      - [Hash](#hash)
+         * [Hash32](#hash32)
+         * [Hash96](#hash96)
+         * [Hash97](#hash97)
+      - [Bytes](#bytes)
+      - [List/Vectors](#listvectors)
+   + [Deserialize/Decode](#deserializedecode)
+      - [uint: 8/16/24/32/64/256](#uint-816243264256-1)
+      - [Address](#address-1)
+      - [Hash](#hash-1)
+         * [Hash32](#hash32-1)
+         * [Hash96](#hash96-1)
+         * [Hash97](#hash97-1)
+      - [Bytes](#bytes-1)
+      - [List/Vectors](#listvectors-1)
+* [Implementations](#implementations)
+
+## About
+
+`SimpleSerialize` was first proposed by Vitalik Buterin as the serialization
+protocol for use in the Ethereum 2.0 Beacon Chain.
+
+The core feature of `ssz` is the simplicity of the serialization with low
+overhead.
+
+## Terminology
+
+| Term         | Definition                                                                                     |
+|:-------------|:-----------------------------------------------------------------------------------------------|
+| `big`        | Big Endian                                                                                     |
+| `byte_order` | Specifies [endianness:](https://en.wikipedia.org/wiki/Endianness) Big Endian or Little Endian. |
+| `len`        | Length/Number of Bytes.                                                                        |
+| `to_bytes`   | Convert to bytes. Should take parameters ``size`` and ``byte_order``.                          |
+| `from_bytes` | Convert form bytes to object. Should take ``bytes`` and ``byte_order``.                        |
+| `value`      | The value to serialize.                                                                        |
+| `rawbytes`   | Raw serialized bytes.                                                                          |
+
+## Constants
+
+| Constant       | Value | Definition                                                              |
+|:---------------|:-----:|:------------------------------------------------------------------------|
+| `LENGTH_BYTES` |   4   | Number of bytes used for the length added before the serialized object. |
+
+
+## Overview
+
+### Serialize/Encode
+
+#### uint: 8/16/24/32/64/256
+
+Convert directly to bytes the size of the int. (e.g. ``uint16 = 2 bytes``)
+
+All integers are serialized as **big endian**.
+
+| Check to perform                 | Code                    |
+|:---------------------------------|:------------------------|
+| Size is a byte integer           | ``int_size % 8 == 0``   |
+| Value is less than max           | ``2**int_size > value`` |
+
+```python
+buffer_size = int_size / 8
```

Comment:
Some of the examples include the "Checks to perform" as `assert` statements, while others do not.

For consistency, I'd recommend including _all_ "check to perform" code snippets in the sample code